pkg/controller/pravegacluster/pravegacluster_controller.go

```diff
@@ -239,12 +239,20 @@ func (r *ReconcilePravegaCluster) deploySegmentStore(p *pravegav1alpha1.PravegaC
 		return err
 	}
 
+	cm := &corev1.ConfigMap{}
+	name := p.Spec.Pravega.SegmentStoreEnvVars
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: p.Namespace}, cm)
 	configMap := pravega.MakeSegmentstoreConfigMap(p)
 	controllerutil.SetControllerReference(p, configMap, r.scheme)
 	err = r.client.Create(context.TODO(), configMap)
 	if err != nil && !errors.IsAlreadyExists(err) {
 		return err
 	}
+	flag := r.compareKeys(cm, configMap)
+	if !flag {
+		log.Println("Rejecting the environment variables provided")
```

Comment:
Please log as Error().

Reply:
Also state which Key caused the clash. Otherwise user does not know what went wrong.